test/framework/machine_helpers.go

```diff
@@ -18,30 +18,104 @@ package framework
 
 import (
 	"context"
+	"fmt"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/pkg/errors"
 
+	"github.com/pkg/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 )
 
-// WaitForMachinesToBeUpgradedInput is the input for WaitForMachinesToBeUpgraded.
-type WaitForMachinesToBeUpgradedInput struct {
+type GetMachinesByMachineDeploymentsInput struct {
+	Lister      Lister
+	ClusterName string
+	Namespace   string
+	Deployment  clusterv1.MachineDeployment
+}
+
+// GetMachinesByMachineDeployments returns Machine objects for a cluster belonging to a machine deployment.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetMachinesByMachineDeployments(ctx context.Context, input GetMachinesByMachineDeploymentsInput) []clusterv1.Machine {
+	opts := byClusterOptions(input.ClusterName, input.Namespace)
```

Comment:
In all the new framework helpers there should be assertions to validate inputs parameters